library/func_misc.py

```diff
@@ -147,3 +147,44 @@ def Call(self, pos_args, named_args):
 
         keys = [value.Str(k) for k in dictionary.keys()]  # type: List[value_t]
         return value.List(keys)
+
+
+class Len(vm._Callable):
+
+    def __init__(self):
+        # type: () -> None
+        pass
+
+    def Call(self, pos_args, named_args):
+        # type: (List[value_t], Dict[str, value_t]) -> value_t
+
+        n_args = len(pos_args)
+        if n_args != 1:
```

Comment:
This check looks good for now, though I also sketched this typed_args.Reader thing that's not implemented

Should discuss on Zulip

When we do proc argument binding, this discussion of the style will probably be more relevant